runtime/codert_vm/cnathelp.cpp

```diff
@@ -885,25 +903,150 @@ old_fast_jitPutFlattenableStaticField(J9VMThread *currentThread)
+				goto done;
+			}
+			currentThread->floatTemp1 = (void*)value; // in case of decompile
+			addr = restoreJITResolveFrame(currentThread, oldPC, false, false);
+			if (NULL != addr) {
+				goto done;
+			}
+			JIT_RETURN_UDATA(value);
+		}
+	}
+done:
+	SLOW_JIT_HELPER_EPILOGUE();
+	return addr;
 }
 
 void* J9FASTCALL
 old_fast_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	OLD_JIT_HELPER_PROLOGUE(2);
+	DECLARE_JIT_PARM(j9object_t, arrayObject, 1);
+	DECLARE_JIT_INT_PARM(index, 2);
+	bool slowPathUsed = false;
+	void *slowPath = NULL;
+	j9object_t value = NULL;
+	U_32 arrayLength = 0;
+	if (NULL == arrayObject) {
+		goto slow;
+	}
+	arrayLength = J9INDEXABLEOBJECT_SIZE(_currentThread, arrayObject);
+	if (index >= arrayLength) {
+		goto slow;
+	}
+	value = (j9object_t) currentThread->javaVM->internalVMFunctions->loadFlattenableArrayElement(currentThread, arrayObject, index, true);
+	if (J9_UNEXPECTED(NULL == value)) {
+		goto slow;
+	}
+	JIT_RETURN_UDATA(value);
+done:
+	return slowPath;
+slow:
+	slowPathUsed = true;
+	currentThread->floatTemp1 = (void *)arrayObject;
+	currentThread->floatTemp2 = *(void **)&index;
+	slowPath = (void*)old_slow_jitLoadFlattenableArrayElement;
+	goto done;
 }
 
 void* J9FASTCALL
 old_slow_jitStoreFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	SLOW_JIT_HELPER_PROLOGUE();
+	j9object_t arrayref = (j9object_t)currentThread->floatTemp1;
+	j9object_t value = (j9object_t)currentThread->floatTemp2;
+	U_32 index = *(U_32 *)&currentThread->floatTemp3;
+	U_32 arrayLength = 0;
+	void *addr = NULL;
+	if (NULL == arrayref) {
+		buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+		addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		arrayLength = J9INDEXABLEOBJECT_SIZE(currentThread, arrayref);
+		if (index >= arrayLength) {
+			buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+			addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGARRAYINDEXOUTOFBOUNDSEXCEPTION, NULL);
+		} else {
+			if (false == jitObjectArrayStoreAllowed(arrayref, value)) {
+				buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+				addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGARRAYSTOREEXCEPTION, NULL);
+			} else {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				J9ArrayClass *arrayrefClass = (J9ArrayClass *) J9OBJECT_CLAZZ(currentThread, arrayref);
+				if (J9_IS_J9CLASS_VALUETYPE(arrayrefClass->componentType) && (NULL == value)) {
+					buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+					addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);;
+				} else
+#endif /* if defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+				{
+					currentThread->javaVM->internalVMFunctions->storeFlattenableArrayElement(currentThread, arrayref, index, value);
```

Comment:
its not actually possible to get here, you can only be in the slow path if you are throwing an exception, since the storing the element cant fail